activation/handler.go

```diff
@@ -202,12 +205,36 @@ func (h *Handler) SyntacticallyValidateAtx(ctx context.Context, atx *types.Activ
 			return nil, fmt.Errorf("initial Post indices included in challenge does not equal to the initial Post indices included in the atx")
 		}
 
+		if atx.CommitmentATX == *types.EmptyATXID {
+			return nil, fmt.Errorf("no prevATX declared, but commitmentATX is missing")
+		}
+
+		if atx.CommitmentATX != h.goldenATXID {
+			commitmentAtx, err := h.cdb.GetAtxHeader(atx.CommitmentATX)
+			if err != nil {
+				return nil, fmt.Errorf("commitment atx not found")
+			}
+			if !atx.PubLayerID.After(commitmentAtx.PubLayerID) {
+				return nil, fmt.Errorf("atx layer (%v) must be after commitment atx layer (%v)", atx.PubLayerID, commitmentAtx.PubLayerID)
+			}
+			if d := atx.PubLayerID.Difference(commitmentAtx.PubLayerID); d > h.layersPerEpoch {
+				return nil, fmt.Errorf("expected distance of one epoch (%v layers) from commitment atx but found %v", h.layersPerEpoch, d)
+			}
+		} else {
+			publicationEpoch := atx.PublishEpoch()
+			if !publicationEpoch.NeedsGoldenPositioningATX() {
```

Comment:
why not just use IsGenesis()? to me it's the same difference.

Reply:
I used this method because it is also used for the positioning ATX; I wasn't aware there are two methods doing essentially the same.

Since `NeedsGoldenPositioningATX` seems to be exclusively used in this method, while `IsGenesis` is used in many different places, I will get rid of it and replace it.